### DISCUSSION TITLE
Continuous Integration

### DISCUSSION THREAD

**Discussion by pitgrap** [pnpm/pnpm#3803] [Q&A] [answered]
Hi, we're using pnpm in CI pipelines with Jenkins. 

In the documentation are different defaults described, for CI and non-CI (e.g. --frozen-lockfile). I would like to know, how pnpm detects a CI environment. And how can I test it? Is it possible to do it manually? What kind of environment flag has to be set?

**Comment by zkochan** [comment, marked as answer]
We use this lib to detect CI: https://www.npmjs.com/package/is-ci
